### PR TITLE
Fix price range display when price is in the lowest or highest bucket (PURCHASE-2970)

### PR DESCRIPTION
**Pull request by starsirius**
Fixes https://artsyproduct.atlassian.net/browse/PURCHASE-2970

When an artwork is set as displaying price range and is in the lowest (`[nil, 1000]`) or highest (`[10_000_000, nil]`) bucket, the price is displayed incorrectly with an exact price, rather than the "Under [price]" or "[price] and up" language.

This fixes it by updating the `priceRangeDisplayText` logic and the `saleMessage` resolver. More details inline.

It's not ideal there if duplicate and inconsistent formatting logic in MP and Gravity that deserves a bigger discussion. This only fixes the immediate issue.

#### Price in the lowest bucket
Screen Shot 2021-09-30 at 11 46 45 PM
Screen Shot 2021-09-30 at 11 47 05 PM

#### Price in the highest bucket
Screen Shot 2021-10-01 at 12 36 03 AM
Screen Shot 2021-10-01 at 12 36 22 AM

### CODE REVIEW

File: src/lib/moneyHelpers.ts [hunk 1]
```diff
@@ -39,49 +39,46 @@ export const isCurrencySupported = (currency: string): boolean => {
 }
 
 /**
- * Builds price display text (e.g. "$100").
+ * Builds price display text (e.g. "US$100").
  */
 export const priceDisplayText = (
-  priceCents: number | [number, number],
```
Comment (starsirius) [inline src/lib/moneyHelpers.ts:45]: Reverting this change from https://github.com/artsy/metaphysics/pull/3393 to make the responsibility clear: formatting a single number.

File: src/lib/moneyHelpers.ts [hunk 2]
```diff
@@ -39,49 +39,46 @@ export const isCurrencySupported = (currency: string): boolean => {
...
 
 /**
- * Builds price range display text (e.g. "$100–$200" or "VUV Vt100–Vt200")..
+ * Builds price range display text (e.g. "US$100–US$200" or "VUV Vt100–Vt200")..
  */
 export const priceRangeDisplayText = (
-  lowerPriceCents: number,
-  higherPriceCents: number,
+  lowerPriceCents: number | null,
+  higherPriceCents: number | null,
```
Comment (starsirius) [inline src/lib/moneyHelpers.ts:57]: [`price_cents`](https://github.com/artsy/gravity/blob/8b1259a48a655aee0d9aa929577c5d08ac8e7e42/app/models/util/for_sale.rb#L208-L220) array from Gravity could return unbounded on either end, for example [`[nil, 100000]`](https://github.com/artsy/gravity/blob/eaf860818d504442f5d666f3cf43667cc5116571/spec/models/domain/artwork_spec.rb#L2037-L2043). This updates the logic of price range display to behave in a similar way as the [money_helper](https://github.com/artsy/money_helper/blob/8fb2110afa122c4ae32ba557000d2c0521fb2bf8/spec/money_helper_spec.rb#L199-L205) gem.

**Review by olerichter00 — APPROVED** [review]
Looks good to me 👍 Feel free to merge whenever you're ready.

**Comment by starsirius** [comment]
Merging and giving this a try! Thanks for the review, @olerichter00 !

**Pull request merged**